crates/webidl/src/main.rs

```diff
@@ -0,0 +1,114 @@
+        if path.extension() != Some(OsStr::new("webidl")) {
+            continue;
+        }
+        source = source
+            .add_file(&path)
+            .with_context(|| format!("reading contents of file \"{}\"", path.display()))?;
+    }
+
+    Ok(source)
+}
+
+fn main() -> Result<()> {
+    #[cfg(feature = "env_logger")]
+    env_logger::init();
+
+    match std::env::args().into_iter().collect::<Vec<_>>().as_slice() {
+        [ _, from, to ] => {
+            let from = path::Path::new(from);
+            let to = path::Path::new(to);
+
+            let source = read_source_from_path(&from.join("enabled"))?;
+            let unstable_source = read_source_from_path(&from.join("unstable"))?;
+
+            let features = match wasm_bindgen_webidl::compile(&source.contents, &unstable_source.contents) {
+                Ok(features) => features,
+                Err(e) => {
+                    if let Some(err) = e.downcast_ref::<wasm_bindgen_webidl::WebIDLParseError>() {
+                        if let Some(pos) = source.resolve_offset(err.0) {
+                            let ctx = format!(
+                                "compiling WebIDL into wasm-bindgen bindings in file \
+                                 \"{}\", line {} column {}",
+                                pos.filename,
+                                pos.line + 1,
+                                pos.col + 1
+                            );
+                            return Err(e.context(ctx));
+                        } else {
+                            return Err(e.context("compiling WebIDL into wasm-bindgen bindings"));
+                        }
+                    }
+                    return Err(e.context("compiling WebIDL into wasm-bindgen bindings"));
+                }
+            };
+
+
+            unwrap_not_found(fs::remove_dir_all(&to)).context("Removing features directory")?;
+            fs::create_dir_all(&to).context("Creating features directory")?;
+
+
+            for (name, feature) in features.iter() {
+                let out_file_path = to.join(format!("gen_{}.rs", name));
+
+                fs::write(&out_file_path, &feature.code)?;
+
+                // run rustfmt on the generated file - really handy for debugging
+                let result = Command::new("rustfmt")
+                    .arg("--edition")
+                    .arg("2018")
+                    .arg(&out_file_path)
+                    .status()
+                    .context(format!("rustfmt on file gen_{}.rs", name))?;
+
+                assert!(result.success(), "rustfmt on file gen_{}.rs", name);
+            }
+
+
+            let binding_file = features.keys().map(|name| {
+                format!("#[cfg(feature = \"{name}\")] mod gen_{name};\n#[cfg(feature = \"{name}\")] pub use gen_{name}::*;", name = name)
+            }).collect::<Vec<_>>().join("\n\n");
+
+            fs::write(to.join("mod.rs"), format!("#![allow(non_snake_case)]\n\n{}", binding_file))?;
```

Comment:
Could rustfmt get executed on this file as well?

Reply:
Is that needed? It's already nicely formatted, and in my opinion running rustfmt would make it larger and harder to read without any benefit.

Reply:
I think it's good for all rust code here to be rustfmt'd, and if we disagree with the style then that's an issue for rustfmt not this repository.